specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2623 @@
+          "readOnly": true,
+          "$ref": "#/definitions/ResourceName"
+        },
+        "type": {
+          "description": "Type of the filesystem; Microsoft.StorageCache/amlFilesystem",
+          "readOnly": true,
+          "type": "string"
+        },
+        "identity": {
+          "$ref": "#/definitions/amlFilesystemIdentity",
+          "description": "The managed identity used by the amlFilesystem, if configured."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of the amlFilesystem.",
+          "properties": {
+            "storageCapacityTiB": {
+              "description": "The size of this filesystem, in TiB.  This may be rounded up.",
+              "type": "number",
+              "format": "float"
+            },
+            "health": {
+              "description": "Health of the amlFilesystem.",
+              "readOnly": true,
+              "$ref": "#/definitions/amlFilesystemHealth"
+            },
+            "availabilityZones": {
+              "type": "array",
+              "items": {
+                "type": "string"
+              },
+              "description": "Availability zones for nodes."
+            },
+            "provisioningState": {
+              "description": "ARM provisioning state, see https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "Cancelled",
+                "Creating",
+                "Deleting",
+                "Updating"
+              ],
+              "x-ms-enum": {
+                "name": "ProvisioningStateType",
+                "modelAsString": true
+              },
+              "type": "string"
+            },
+            "subnet": {
+              "description": "Subnet used for the amlFilesystem.",
+              "$ref": "#/definitions/URLString"
+            },
+            "encryptionSettings": {
+              "$ref": "#/definitions/amlFilesystemEncryptionSettings",
+              "description": "Specifies encryption settings of the amlFilesystem."
+            },
+            "maintenanceWindow": {
+              "description": "Start time of a 30 minute weekly maintenance window",
+              "type" : "string",
+              "format" : "date-time"
+            },
+            "maintenanceStatus": {
+              "description": "Maintenance status of the amlFilesystem.",
+              "$ref": "#/definitions/amlFilesystemMaintenanceStatus"
+            },
+            "hsmSettings": {
+              "$ref": "#/definitions/amlFilesystemHSMSettings",
+              "description": "Specifies HSM settings of the amlFilesystem."
+            },
+            "namespaceHydrationStatus": {
+              "description" : "Namespace Hydration Status during creation",
+              "$ref": "#/definitions/amlFilesystemHydrationStatus"
```

Comment:
It seems inconsistent to have "namespaceHydrationStatus" and "FilesystemHydrationStatus". There is namespace hydration and data hydration. I'd expect "FilesystemHydrationStatus" to include both.

Reply:
renamed object based on a later comment to amlFilesystemNamespaceHydrationStatus.
Fixed, see on next commit.